Source/WebCore/workers/service/background-fetch/BackgroundFetchManager.cpp

```diff
@@ -28,31 +28,147 @@
 
 #if ENABLE(SERVICE_WORKER)
 
-#include "ServiceWorkerRegistration.h"
+#include "BackgroundFetchInformation.h"
+#include "BackgroundFetchRequest.h"
+#include "FetchRequest.h"
+#include "JSBackgroundFetchRegistration.h"
+#include "SWClientConnection.h"
+#include "ServiceWorkerProvider.h"
 
 namespace WebCore {
 
-BackgroundFetchManager::BackgroundFetchManager(ServiceWorkerRegistration&)
+BackgroundFetchManager::BackgroundFetchManager(ServiceWorkerRegistration& registration)
+    : m_identifier(registration.identifier())
 {
 }
 
 BackgroundFetchManager::~BackgroundFetchManager()
 {
 }
 
-void BackgroundFetchManager::fetch(ScriptExecutionContext&, const String&, Requests&&, BackgroundFetchOptions&&, DOMPromiseDeferred<IDLInterface<BackgroundFetchRegistration>>&& promise)
+static ExceptionOr<Vector<Ref<FetchRequest>>> buildBackgroundFetchRequests(ScriptExecutionContext& context, BackgroundFetchManager::Requests&& backgroundFetchRequests)
 {
-    promise.reject(Exception { NotSupportedError });
+    if (std::holds_alternative<RefPtr<FetchRequest>>(backgroundFetchRequests)) {
```

Comment:
Any reason we're not using WTF::switchOn() instead of std::holds_alternative + std::get?

Reply:
I'll switch to switch ;)